difference.go

```diff
@@ -32,11 +34,11 @@ type objectDifference func(string, string, os.FileMode, int64, time.Time) (diffe
 type differType string
 
 const (
-	differInSize  differType = "size"          // differs in size
-	differInTime             = "time"          // differs in time
-	differInFirst            = "only-in-first" // only on first source
-	differInType             = "type"          // differs in type, exfile/directory
-	differInNone             = ""              // does not differ
+	differInSize   differType = "size"           // differs in size
```

Comment:
Let's change this from string to 'iota' style 

```
type differType int 

const (
     differInNone differType = iota
     differInSize
     differInType
     differInFirst
     differInSecond
)
```

This helps in not passing around strings and also simpler matching techniques.